File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink/20260831_075035/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    { description: "Quantity below 10 (9 units) - no discount", price: 100, quantity: 9, expected: 100 * 9 },
    { description: "Quantity at 10 (boundary) - 10% discount", price: 100, quantity: 10, expected: 100 * 10 * 0.9 },
    { description: "Quantity at 49 (boundary) - 10% discount", price: 100, quantity: 49, expected: 100 * 49 * 0.9 },
    { description: "Quantity at 50 (boundary) - 25% discount", price: 100, quantity: 50, expected: 100 * 50 * 0.75 },
];

let allPassed = true;

for (const tc of testCases) {
    const got = calculateDiscount(tc.price, tc.quantity);
    // Compare with a small tolerance for floating point
    const isClose = Math.abs(got - tc.expected) < 1e-9;
    if (isClose) {
        console.log(`PASS: ${tc.description}`);
    } else {
        console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${got})`);
        allPassed = false;
    }
}

if (allPassed) {
    console.log("\nAll tests passed!");
} else {
    console.log("\nSome tests failed.");
    process.exit(1);
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink/20260831_075035/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def test_calculate_discount():
    test_cases = [
        (10.0, 9, 90.0, "Quantity 9 (below 10, no discount)"),
        (10.0, 10, 90.0, "Quantity 10 (boundary, 10% off)"),
        (10.0, 49, 441.0, "Quantity 49 (boundary, 10% off)"),
        (10.0, 50, 375.0, "Quantity 50 (boundary, 25% off)"),
    ]
    
    for price, quantity, expected, description in test_cases:
        got = calculate_discount(price, quantity)
        if abs(got - expected) < 1e-9:
            print(f"PASS: {description}")
        else:
            print(f"FAIL: {description} (expected: {expected}, got: {got})")


test_calculate_discount()